CMSgov/onemac
# Oy2 22595 - always use seatool proposed effective date to backfill event data

Story: https://qmacbis.atlassian.net/browse/OY2-22595
Endpoint: See github-actions bot comment

### Details

Due to order of data processing when building packages, older ONEMAC events were being used to populate proposed effective date rather than the newer SEATOOL event.

### Changes

- Updated logic in buildAnyPackage

### Implementation Notes

- First attempt was to backfill all data from previous package record; however since ONEMAC records are processed first due to sk order the ONEMAC records were being used to populate proposed effective date and the newest status chane ONEMAC event had a later event timestamp than the SEATOOL event. Therefore the SEATOOL event was not used. For now since editing of a onemac package is not possible, assume any data other than status from SEATOOL can always be applied (currently only proposed effective date) regardless if its the latest event.

### Test Plan

1. Submit a package.
2. Submit a seatool event (or insert event into DB) to set a proposed effective date and PENDING status on the submitted package.
3. Withdraw package.
4. Verify pakcage retains proposed effective date set in seatool event.
